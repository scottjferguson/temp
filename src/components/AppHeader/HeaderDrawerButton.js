import React from 'react';
import PropTypes from 'prop-types';
import { withNavigation } from 'react-navigation';
import { DrawerActions } from 'react-navigation-drawer';
import { Button, Icon } from 'native-base';

const HeaderDrawerButton = ({ navigation }) => {
  return (
    <Button
      transparent
      onPress={() => navigation.dispatch(DrawerActions.toggleDrawer({}))}>
      <Icon
        active
        type="SimpleLineIcons"
        name="menu"
        style={{ fontSize: 26 }}
      />
    </Button>
  );
};

HeaderDrawerButton.propTypes = {
  navigation: PropTypes.shape({ dispatch: PropTypes.func.isRequired }),
};

export default withNavigation(HeaderDrawerButton);
