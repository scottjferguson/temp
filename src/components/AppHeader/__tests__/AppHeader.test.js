import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AppHeader from '../index';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';

describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders correctly', async () => {
    const props = {
      navigation: { navigate: jest.fn(), dispatch: jest.fn() },
      style: {},
      title: 'Expenses',
      titleSuffix: 'June',
      subTitle: 'Manage your money',
      displayAvatar: true,
      displayLogo: true,
    };
    const tree = renderer.create(<AppHeader {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
