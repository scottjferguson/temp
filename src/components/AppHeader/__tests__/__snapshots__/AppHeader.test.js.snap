// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {},
    ]
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      hasTabs={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "transparent",
          "borderBottomWidth": null,
          "elevation": 0,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": 0,
        }
      }
      transparent={true}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 4,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "resizeMode": "contain",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                },
                Object {
                  "borderRadius": 20,
                  "height": 40,
                  "width": 40,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#FFF",
              "fontFamily": "Roboto_light",
              "fontSize": 26,
            },
          ]
        }
        uppercase={false}
      >
        Expenses
      </Text>
      <Text
        note={true}
        style={
          Array [
            Object {
              "color": "#a7a7a7",
              "fontFamily": "System",
              "fontSize": 14,
            },
            Object {
              "color": "#FFF",
              "fontFamily": "Roboto_light",
              "fontSize": 26,
              "fontWeight": "100",
              "opacity": 0.8,
            },
          ]
        }
        uppercase={false}
      >
         June
      </Text>
    </View>
    <Text
      note={true}
      style={
        Array [
          Object {
            "color": "#a7a7a7",
            "fontFamily": "System",
            "fontSize": 14,
          },
          Object {
            "color": "#FFF",
            "fontFamily": "Roboto_light",
            "fontSize": 24,
            "fontWeight": "100",
            "opacity": 0.9,
            "paddingTop": 10,
          },
        ]
      }
      uppercase={false}
    >
      Manage your money
    </Text>
  </View>
</View>
`;
